### ISSUE TITLE
Gaia hangs on IME input

### ISSUE THREAD

**Issue by reinux** [bryanedds/Nu#541]
Trying to type in Japanese using IME first seems to cause diacritics on Latin letters as opposed to Japanese, and then eventually causes a crash.

IMGUI [seems to support IME](https://github.com/ocornut/imgui/issues/471) so I'm neither worried nor in a hurry, but it might be good to at least work around it (maybe even by disabling IME) to prevent crashing.

![image](https://github.com/bryanedds/Nu/assets/5127297/fe446f60-3936-4f0f-bb6c-4ee55229c8bc)

**Comment by bryanedds** [comment]
Is there a quick way to duplicate this on a standard US keyboard?

**Comment by reinux** [comment]
On Windows, you can add new keyboards in the Language & region settings.

![image](https://github.com/bryanedds/Nu/assets/5127297/33331e0f-fe74-411c-947a-dafca75ff701)

Once you have it installed, you can switch to Japanese on the bottom right of the Taskbar. Start + Space to switch between different keyboards, but Japanese also has a legacy English mode built in, so you may also need to do Alt + Backtick (`) to finally start typing.

After that, you can type as you normally would if you were typing Japanese in English, and then hit space for an (increasingly shoddy, AI-driven) autocomplete/autocorrect menu.

**Comment by bryanedds** [comment]
A couple notes:

I think SDL is also part of this input pipeline as well, since we get the text from the SDL_TEXTINPUT event.

Also, I'm using the standard, out-of-box ImGui font which I'm pretty sure only shows non-extended characters.

**Comment by reinux** [comment]
I think in an ideal situation it would be using fallback fonts given by the system, so that Latin/ASCII characters are rendered using ImGui's monospace font, and everything else by the system font.

Or, something that might get a lot of people excited is one of the SNES style fonts, like the 8x8px [Misaki](https://littlelimit.net/misaki.htm) or the [10x10 and 12x12 M+ BITMAP](http://mplus-fonts.osdn.jp/mplus-bitmap-fonts/index.html#features). Both have simple and very permissive licenses:

http://mplus-fonts.osdn.jp/mplus-bitmap-fonts/index.html#license
https://littlelimit.net/font.htm#license

**Comment by bryanedds** [comment]
This may be related - https://github.com/ocornut/imgui/issues/5535

**Comment by bryanedds** [comment]
Here's some additional context -

https://discourse.libsdl.org/t/can-i-use-sdl-disable-windows-ime-and-ignore-other-input-at-the-same-time-without-hacks/24468

https://github.com/Rust-SDL2/rust-sdl2/issues/1041

As a workaround, I'm trying to disable windows IME in SDL2, but I'm not yet able to figure out how via SDL2-CS -

https://github.com/search?q=repo%3Aflibitijibibo%2FSDL2-CS+IME

**Comment by bryanedds** [comment]
Maybe related - https://github.com/ocornut/imgui/pull/7461

**Comment by bryanedds** [comment]
whole tag of maybe related - https://github.com/ocornut/imgui/pulls?q=is%3Apr+is%3Aopen+label%3Aime

**Comment by bryanedds** [comment]
I added japanese lang support to my Windows 10 install and typed various japanese characters in Gaia, but was still unable to trigger a hang. Perhaps the OS call that led to the hang has been patched in a Windows update?

I asked rei to give me a specific japanese phrase to type so we can at least verify what seems to be a lack of hanging.

**Comment by bryanedds** [comment]
Yes, I'm suspecting the hang itself is fixed with a windows patch.
